controllers/nutanixcluster_controller_test.go

```diff
@@ -530,34 +535,23 @@ func TestNutanixClusterReconciler(t *testing.T) {
 			})
 		})
 
-		Context("Delete credentials ref reconcile failed: PrismCentral Info is null", func() {
+		Context("NutanixCluster creation failed: PrismCentral Info is null", func() {
 			It("Should not return error", func() {
 				ctx := context.Background()
-				reconciler := &NutanixClusterReconciler{
-					Client: k8sClient,
-					Scheme: runtime.NewScheme(),
-				}
 
 				ntnxCluster := &infrav1.NutanixCluster{
 					ObjectMeta: metav1.ObjectMeta{
 						Name:      "test",
 						Namespace: "default",
 					},
 					Spec: infrav1.NutanixClusterSpec{
-						PrismCentral: nil,
+						ControlPlaneEndpoint: capiv1.APIEndpoint{},
+						PrismCentral:         nil,
 					},
 				}
 
 				// Create the NutanixCluster object
-				g.Expect(k8sClient.Create(ctx, ntnxCluster)).To(Succeed())
-				defer func() {
-					err := k8sClient.Delete(ctx, ntnxCluster)
-					Expect(err).NotTo(HaveOccurred())
-				}()
-
-				// Reconile Delete credential ref
-				err := reconciler.reconcileCredentialRefDelete(ctx, ntnxCluster)
-				g.Expect(err).NotTo(HaveOccurred())
```

Comment:
why are we overwriting this test and reducing it's scope?
removing this caused a drop in code coverage leading the codecov job to fail https://app.codecov.io/gh/nutanix-cloud-native/cluster-api-provider-nutanix/pull/400/indirect-changes

Reply:
this is the error from capx-controller log when we dont set prismCentralEndpoint 
<pre>
{"level":"info","ts":"2024-04-08T23:01:59Z","msg":"Reconciling the NutanixMachine.","controller":"nutanixmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"NutanixMachine","NutanixMachine":{"name":"mycluster-without-topology-mt-0-kfwvs","namespace":"ns-topology"},"namespace":"ns-topology","name":"mycluster-without-topology-mt-0-kfwvs","reconcileID":"c15c942a-ce12-4076-b638-cc3febbcd0d3"}
{"level":"info","ts":"2024-04-08T23:01:59Z","msg":"Fetched the owner Machine: mycluster-without-topology-wmd-29wcn-vsksk","controller":"nutanixmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"NutanixMachine","NutanixMachine":{"name":"mycluster-without-topology-mt-0-kfwvs","namespace":"ns-topology"},"namespace":"ns-topology","name":"mycluster-without-topology-mt-0-kfwvs","reconcileID":"c15c942a-ce12-4076-b638-cc3febbcd0d3"}
{"level":"info","ts":"2024-04-08T23:01:59Z","msg":"[WARNING] prismCentral attribute was not set on NutanixCluster mycluster-without-topology in namespace ns-topology. Defaulting to CAPX manager credentials","controller":"nutanixmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"NutanixMachine","NutanixMachine":{"name":"mycluster-without-topology-mt-0-kfwvs","namespace":"ns-topology"},"namespace":"ns-topology","name":"mycluster-without-topology-mt-0-kfwvs","reconcileID":"c15c942a-ce12-4076-b638-cc3febbcd0d3"}
{"level":"info","ts":"2024-04-08T23:01:59Z","msg":"Warning: Reconciler returned both a non-zero result and a non-nil error. The result will always be ignored if the error is non-nil and the non-nil error causes reqeueuing with exponential backoff. For more details, see: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler","controller":"nutanixmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"NutanixMachine","NutanixMachine":{"name":"mycluster-without-topology-mt-0-kfwvs","namespace":"ns-topology"},"namespace":"ns-topology","name":"mycluster-without-topology-mt-0-kfwvs","reconcileID":"c15c942a-ce12-4076-b638-cc3febbcd0d3"}
{"level":"error","ts":"2024-04-08T23:01:59Z","msg":"Reconciler error","controller":"nutanixmachine","controllerGroup":"infrastructure.cluster.x-k8s.io","controllerKind":"NutanixMachine","NutanixMachine":{"name":"mycluster-without-topology-mt-0-kfwvs","namespace":"ns-topology"},"namespace":"ns-topology","name":"mycluster-without-topology-mt-0-kfwvs","reconcileID":"c15c942a-ce12-4076-b638-cc3febbcd0d3","error":"client auth error: failed to get management endpoint object: the PrismCentral credentials data is not set","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:227"}
</pre>